src/cmd-cloud-prune

```diff
@@ -414,5 +437,71 @@ def prune_build(s3_client, bucket, prefix, build_id, dry_run):
                 raise Exception(f"Error pruning {build_id}: {e.response['Error']['Message']}")
 
 
+def get_container_tags(meta_json, stream):
+    container_tags = []
+    base_oscontainer = meta_json.get("base-oscontainer")
+    if base_oscontainer:
+        tags = base_oscontainer.get("tags", [])
+        # Only include tags that do not match the stream i.e. moving tags
+        filtered_tags = [tag for tag in tags if tag != stream]
+        if filtered_tags:
+            container_tags = filtered_tags
+    return container_tags
+
+
+def prune_containers(containers_config):
+    barrier_releases = set()
+    # Get the update graph for stable streams
+    if containers_config["stream"] in ['stable', 'testing', 'next']:
+        update_graph = get_update_graph(containers_config["stream"])['releases']
+        # Keep only the barrier releases
+        barrier_releases = set([release["version"] for release in update_graph if "barrier" in release])
+
+    for tag in containers_config["container_tags"]:
+        if tag in STREAMS:
+            continue
+        if tag in barrier_releases:
+            print(f"Release {tag} is a barrier release, keeping.")
+            continue
+        if containers_config["dry_run"]:
+            print(f"Would prune image {containers_config["repository_url"]}:{tag}")
+        else:
+            skopeo_delete(containers_config["repository_url"], tag, containers_config["registry_auth_file"])
+
+
+def get_update_graph(stream):
+    url = f"https://builds.coreos.fedoraproject.org/updates/{stream}.json"
+    r = requests.get(url, timeout=5)
+    if r.status_code != 200:
+        raise Exception(f"Could not download update graph for {stream}. HTTP {r.status_code}")
+    return r.json()
+
+
+def skopeo_inspect(repo, image, auth):
+    skopeo_args = ["skopeo", "inspect", f"docker://{repo}:{image}"]
+    if auth:
+        skopeo_args.extend(["--authfile", auth])
+    try:
+        subprocess.check_output(skopeo_args, stderr=subprocess.STDOUT)
+        return True  # Inspection succeeded
+    except subprocess.CalledProcessError as e:
+        print("Inspection failed:", e.output.decode("utf-8"))
+        return False  # Inspection failed
```

Comment:
hmm. I wonder if we can make this better.. i.e. right now this could fail for any reason, but is there a way to only return `False` here if the call was successful, but the tag just doesn't exist?

Reply:
it [looks like](https://github.com/containers/skopeo/blob/main/docs/skopeo.1.md#exit-status) we can check the exitcode for generic error versus image doesn't exist:

```
[dustymabe@hattop ~]$ skopeo inspect docker://quay.io/fedora/fedora-coreos:stabl
FATA[0000] Error parsing image name "docker://quay.io/fedora/fedora-coreos:stabl": reading manifest stabl in quay.io/fedora/fedora-coreos: manifest unknown 
[dustymabe@hattop ~]$ echo $?
2
[dustymabe@hattop ~]$ skopeo inspect docker://quayd.io/fedora/fedora-coreos:stabl
FATA[0000] Error parsing image name "docker://quayd.io/fedora/fedora-coreos:stabl": pinging container registry quayd.io: Get "https://quayd.io/v2/": dial tcp: lookup quayd.io: no such host 
[dustymabe@hattop ~]$ echo $?
1
```

Reply:
resolved in new upload